smartmeter_datacollector/smartmeter/siemens_td3511.py

```diff
@@ -0,0 +1,257 @@
+                self._callback(data)
+            except asyncio.exceptions.TimeoutError:
+                LOGGER.debug("Finished reading dataset F001")
+                self._callback(SiemensSerialReader.TERMINATION_FLAG)
+                break
+        return
+
+    async def _get_f009_dataset(self):
+        # Read dataset F009
+        self.timestamp = datetime.now(timezone.utc)
+        await self._serial.write_async(bytes.fromhex('0152320246303039031E0D0A'))
+        data: bytes = await self._serial.readline_async(size=-1)
+        self._callback(data)
+        while True:
+            try:
+                data: bytes = await asyncio.wait_for(self._serial.readline_async(size=-1), timeout=0.2)
+                self._callback(data)
+            except asyncio.exceptions.TimeoutError:
+                LOGGER.debug("Finished reading dataset F009")
+                self._callback(SiemensSerialReader.TERMINATION_FLAG)
+                break
+        return
+
+
+@dataclass
+class RegisterDataPoint:
+    obis: str
+    data_point_type: MeterDataPointType
+    scaling: float = 1.0
+
+
+DEFAULT_REGISTER_MAPPING = [
+    RegisterDataPoint("1.7.0", MeterDataPointTypes.ACTIVE_POWER_P.value, 1000),
+    RegisterDataPoint("2.7.0", MeterDataPointTypes.ACTIVE_POWER_N.value, 1000),
+    RegisterDataPoint("3.7.0", MeterDataPointTypes.REACTIVE_POWER_P.value, 1000),
+    RegisterDataPoint("4.7.0", MeterDataPointTypes.REACTIVE_POWER_N.value, 1000),
+    RegisterDataPoint("14.7", MeterDataPointTypes.NET_FREQUENCY.value),
+
+    RegisterDataPoint("31.7", MeterDataPointTypes.CURRENT_L1.value),
+    RegisterDataPoint("32.7", MeterDataPointTypes.VOLTAGE_L1.value),
+    RegisterDataPoint("81.7.4", MeterDataPointTypes.ANGLE_UI_L1.value, 3.141592653589793 / 180),
+
+    RegisterDataPoint("51.7", MeterDataPointTypes.CURRENT_L2.value),
+    RegisterDataPoint("52.7", MeterDataPointTypes.VOLTAGE_L2.value),
+    RegisterDataPoint("81.7.15", MeterDataPointTypes.ANGLE_UI_L2.value, 3.141592653589793 / 180),
+
+    RegisterDataPoint("71.7", MeterDataPointTypes.CURRENT_L3.value),
+    RegisterDataPoint("72.7", MeterDataPointTypes.VOLTAGE_L3.value),
+    RegisterDataPoint("81.7.26", MeterDataPointTypes.ANGLE_UI_L3.value, 3.141592653589793 / 180),
+
+    RegisterDataPoint("1.8.0", MeterDataPointTypes.ACTIVE_ENERGY_P.value, 1000),
+    RegisterDataPoint("1.8.1", MeterDataPointTypes.ACTIVE_ENERGY_P_T1.value, 1000),
+    RegisterDataPoint("1.8.2", MeterDataPointTypes.ACTIVE_ENERGY_P_T2.value, 1000),
+    RegisterDataPoint("2.8.0", MeterDataPointTypes.ACTIVE_ENERGY_N.value, 1000),
+    RegisterDataPoint("2.8.1", MeterDataPointTypes.ACTIVE_ENERGY_N_T1.value, 1000),
+    RegisterDataPoint("2.8.2", MeterDataPointTypes.ACTIVE_ENERGY_N_T2.value, 1000),
+    RegisterDataPoint("3.8.1", MeterDataPointTypes.REACTIVE_ENERGY_P.value, 1000),
+    RegisterDataPoint("4.8.1", MeterDataPointTypes.REACTIVE_ENERGY_N.value, 1000),
+]
+
+
+class SiemensParser():
```

Comment:
I kept the parser in the same file, because I did't want to clutter your file structure. So far the parser is only used for Siemens TD3511. I guess it will make more sense to extract when used for other meters as well (maybe up to come). If you want me to do it anyway, I'm happy to do so...

Reply:
No, that's fine like this. I like your proposition 👍